persistence/trees/trees.go

```diff
@@ -23,6 +24,9 @@ import (
 	"github.com/pokt-network/smt"
 )
 
+// declare a logger for use by the tree store
+var logger = pocketLogger.Global.CreateLoggerForModule(modules.TreeStoreModuleName)
```

Comment:
Why not make it part of treestore to follow the same pattern we use for other modules/structs? https://github.com/pokt-network/pocket/wiki/Logger-Readme#logger-initialization

Seems like `treeStore` is available everywhere you use this global variable.

Reply:
because the newMemStateTrees function wouldn't have access to it otherwise, and it doesn't need to be used outside of this module. But I could refactor it onto the treeStore struct with minimal changes.

Reply:
> But I could refactor it onto the treeStore struct with minimal changes.

I would advocate for this (if possible) just for consistency alongside the other local loggers we create

Reply:
I rolled the logger into a module option instead, so it's passed from the persistence module at creation time. 👍